src/native/corehost/browserhost/libBrowserHost.js

```diff
@@ -0,0 +1,536 @@
+            ENVIRONMENT_IS_WORKER: table[7],
+            ENVIRONMENT_IS_SIDECAR: table[8],
+            browserHostResolveMain: table[9],
+            browserHostRejectMain: table[10],
+            getRunMainPromise: table[11],
+        };
+        const jsEngineLocal = {
+            IS_NODE: loaderExportsLocal.ENVIRONMENT_IS_NODE(),
+            IS_SHELL: loaderExportsLocal.ENVIRONMENT_IS_SHELL(),
+            IS_WEB: loaderExportsLocal.ENVIRONMENT_IS_WEB(),
+            IS_WORKER: loaderExportsLocal.ENVIRONMENT_IS_WORKER(),
+            IS_SIDECAR: loaderExportsLocal.ENVIRONMENT_IS_SIDECAR(),
+        };
+        Object.assign(loaderExports, loaderExportsLocal);
+        Object.assign(logger, loggerLocal);
+        Object.assign(assert, assertLocal);
+        Object.assign(jsEngine, jsEngineLocal);
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    function runtimeExportsToTable(map) {
+        return [];
+    }
+    function runtimeExportsFromTable(table, runtime) {
+        Object.assign(runtime, {});
+    }
+    function hostNativeExportsToTable(map) {
+        return [
+            map.registerDllBytes,
+            map.isSharedArrayBuffer,
+        ];
+    }
+    function hostNativeExportsFromTable(table, native) {
+        const nativeLocal = {
+            registerDllBytes: table[0],
+            isSharedArrayBuffer: table[1],
+        };
+        Object.assign(native, nativeLocal);
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    function interopJavaScriptNativeExportsToTable(map) {
+        return [];
+    }
+    function interopJavaScriptNativeExportsFromTable(table, interop) {
+        const interopLocal = {};
+        Object.assign(interop, interopLocal);
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    function nativeBrowserExportsToTable(map) {
+        return [];
+    }
+    function nativeBrowserExportsFromTable(table, interop) {
+        const interopLocal = {};
+        Object.assign(interop, interopLocal);
+    }
+
+    // Licensed to the .NET Foundation under one or more agreements.
+    // The .NET Foundation licenses this file to you under the MIT license.
+
+    // Licensed to the .NET Foundation under one or more agreements.
+    // The .NET Foundation licenses this file to you under the MIT license.
+    // see also `reserved` in `rollup.config.defines.js`
+
+    // Licensed to the .NET Foundation under one or more agreements.
+    // The .NET Foundation licenses this file to you under the MIT license.
+    const loadedAssemblies = {};
+    function registerDllBytes(bytes, asset) {
+        const sp = Module.stackSave();
+        try {
+            const sizeOfPtr = 4;
+            const ptrPtr = Module.stackAlloc(sizeOfPtr);
+            if (Module._posix_memalign(ptrPtr, 16, bytes.length)) {
+                throw new Error("posix_memalign failed");
+            }
+            const ptr = Module.HEAPU32[ptrPtr >> 2];
```

Comment:
Don't we need to use `>>>` to treat the ptr as unsigned? Or am I misremembering

Reply:
done